### PR TITLE
Added cancel download option

### PR DESCRIPTION
**Pull request by Externoak**
As stated in title.

### CODE REVIEW

File: src/components/AircraftSection/index.tsx [hunk 1]
```diff
@@ -37,14 +36,17 @@ type indexProps = {
...
-    const [selectedVariant, setSelectedVariant] = useState<ModVariant>(props.mod.variants[0]);
+    const [selectedVariant] = useState<ModVariant>(props.mod.variants[0]);
     const [selectedTrack, setSelectedTrack] = useState<ModTrack>(props.mod.variants[0]?.tracks[0]);
-    const [needsUpdate, setNeedsUpdate] = useState<Boolean>(false);
+    const [needsUpdate, setNeedsUpdate] = useState<boolean>(false);
 
     useEffect(() => {
-        checkForUpdates()
-    }, [])
+        checkForUpdates().then();
```
Comment (ilyeshammadi) [inline src/components/AircraftSection/index.tsx:48]: **nitpick:** No need for the `then()` to trigger the promise.

File: src/components/AircraftSection/index.tsx [hunk 2]
```diff
@@ -78,15 +80,20 @@ const index: React.FC<indexProps> = (props: indexProps) => {
...
+            });
             props.setIsDownloading(true);
             props.setIsUpdated(false);
             const msfs_package_dir = settings.get('mainSettings.msfsPackagePath');
 
-            const deleteHandle = fs.rmdir(msfs_package_dir + props.mod.targetDirectory + '\\', {recursive: true}, () => {
-                console.log("Deleted original file");
-            });
+            // I wouldn't delete the A32NX folder can run into issues and if patches are applied only patches files will be available
+            // const deleteHandle = fs.rmdir(msfs_package_dir + props.mod.targetDirectory + '\\', {recursive: true}, () => {
```
Comment (ilyeshammadi) [inline src/components/AircraftSection/index.tsx:94]: **nitpick:** If this is not needed, then can be deleted.

File: src/components/AircraftSection/index.tsx [hunk 3]
```diff
@@ -101,23 +108,40 @@ const index: React.FC<indexProps> = (props: indexProps) => {
...
 
-                if (lastPercentFloor !== newPercentFloor) {
-                    lastPercentFloor = newPercentFloor;
-                    props.setDownloadPercentage(lastPercentFloor);
+                    if (lastPercentFloor !== newPercentFloor) {
+                        lastPercentFloor = newPercentFloor;
+                        props.setDownloadPercentage(lastPercentFloor);
+                    }
+                } catch (e) {
+                    if (e.name == 'AbortError') {
```
Comment (ilyeshammadi) [inline src/components/AircraftSection/index.tsx:130]: **nitpick:** I think '===' is safer here.

File: src/components/AircraftSection/index.tsx [hunk 4]
```diff
@@ -130,7 +154,7 @@ const index: React.FC<indexProps> = (props: indexProps) => {
             if (typeof msfs_package_dir === "string") {
                 const zipFile = new Zip(compressedBuffer);
 
-                await deleteHandle;
+                // await deleteHandle;
```
Comment (ilyeshammadi) [inline src/components/AircraftSection/index.tsx:157]: **nitpick:** Can be cleaned 🧹

File: src/components/AircraftSection/index.tsx [hunk 5]
```diff
@@ -154,12 +178,20 @@ const index: React.FC<indexProps> = (props: indexProps) => {
...
         }
     }
+    function cancelDownload() {
+        if (props.isDownloading) {
+            console.log('Cancel download');
+            controller.abort();
+        }
+    }
+
+    // useEffect(() => {});
```
Comment (ilyeshammadi) [inline src/components/AircraftSection/index.tsx:194]: **nitpick:** 🧹

**Review by ilyeshammadi — CHANGES_REQUESTED** [review]

**Review by ZigTag — APPROVED** [review]
LGTM

**Pull request merged**